src/gstring.rs

```diff
@@ -206,9 +206,9 @@ impl From<Box<str>> for GString {
     }
 }
 
-impl From<&'static str> for GString {
+impl<'a> From<&'a str> for GString {
```

Comment:
You need to add a lifetime to GString too then,otherwise this is unsafe

Reply:
Hum this would need another gir codegen round it seems

Reply:
Yes, but as we need to copy anyway (to add the `NUL` terminator), there's not much point in this. The previous code was wrong, it omitted the `NUL` terminator so `to_glib_none()` would be broken.